packages/editor/src/store/actions.js

```diff
@@ -432,175 +359,15 @@ export function __experimentalOptimisticUpdatePost( edits ) {
-	// if the autosave is applied as a revision.
-	yield dispatch(
-		STORE_KEY,
-		'__experimentalOptimisticUpdatePost',
-		toSend
-	);
-
-	let path = `/wp/v2/${ postType.rest_base }/${ post.id }`;
-	let method = 'PUT';
-	if ( isAutosave ) {
-		const currentUser = yield select( 'core', 'getCurrentUser' );
-		const currentUserId = currentUser ? currentUser.id : undefined;
-		const autosavePost = yield select( 'core', 'getAutosave', post.type, post.id, currentUserId );
-		const mappedAutosavePost = mapValues( pick( autosavePost, AUTOSAVE_PROPERTIES ), getPostRawValue );
-
-		// Ensure autosaves contain all expected fields, using autosave or
-		// post values as fallback if not otherwise included in edits.
-		toSend = {
-			...pick( post, AUTOSAVE_PROPERTIES ),
-			...mappedAutosavePost,
-			...toSend,
-		};
-		path += '/autosaves';
-		method = 'POST';
-	} else {
-		yield dispatch(
-			'core/notices',
-			'removeNotice',
-			SAVE_POST_NOTICE_ID
-		);
-		yield dispatch(
-			'core/notices',
-			'removeNotice',
-			'autosave-exists'
-		);
-	}
-
-	try {
-		const newPost = yield apiFetch( {
-			path,
-			method,
-			data: toSend,
-		} );
-
-		if ( isAutosave ) {
-			yield dispatch( 'core', 'receiveAutosaves', post.id, newPost );
-		} else {
-			yield dispatch( STORE_KEY, 'resetPost', newPost );
-		}
-
-		yield dispatch(
-			STORE_KEY,
-			'__experimentalRequestPostUpdateSuccess',
-			{
-				previousPost: post,
-				post: newPost,
-				options,
-				postType,
-				// An autosave may be processed by the server as a regular save
-				// when its update is requested by the author and the post was
-				// draft or auto-draft.
-				isRevision: newPost.id !== post.id,
-			}
-		);
-
-		const notifySuccessArgs = getNotificationArgumentsForSaveSuccess( {
-			previousPost: post,
-			post: newPost,
-			postType,
-			options,
-		} );
-		if ( notifySuccessArgs.length > 0 ) {
-			yield dispatch(
-				'core/notices',
-				'createSuccessNotice',
-				...notifySuccessArgs
-			);
-		}
-	} catch ( error ) {
-		yield dispatch(
-			STORE_KEY,
-			'__experimentalRequestPostUpdateFailure',
-			{ post, edits, error, options }
-		);
-		const notifyFailArgs = getNotificationArgumentsForSaveFail( {
-			post,
-			edits,
-			error,
-		} );
-		if ( notifyFailArgs.length > 0 ) {
-			yield dispatch(
-				'core/notices',
-				'createErrorNotice',
-				...notifyFailArgs
-			);
-		}
-	}
+	// TODO: If edits can be managed for multiple core data entities, the save
+	// procedure should be reconsidered as persisting all outstanding edits, not
+	// just for the single post being edited.
+	yield __experimentalRequestPostUpdateStart( options );
+	const postType = yield select( 'core/editor', 'getCurrentPostType' );
+	const postId = yield select( 'core/editor', 'getCurrentPostId' );
+	const edits = yield select( 'core', 'getEntityRecordEdits', 'postType', postType, postId );
+	const record = { id: postId, ...edits };
+	yield dispatch( 'core', 'saveEntityRecord', 'postType', postType, record );
```

Comment:
Shouldn't this just persist the current edits if none are provided?

Reply:
> Shouldn't this just persist the current edits if none are provided?

Could you clarify what you mean by this? The `edits` state should ideally reflect the minimal difference between the server-persisted copy, so what's sent here is only those subset of properties.

Reply:
Yeah, but is `saveEntityRecord` ever going to be called with anything other than the result of `getEntityRecordEdits`? I think that can go inside `saveEntityRecord`, at least as a default value.

Reply:
> Yeah, but is `saveEntityRecord` ever going to be called with anything other than the result of `getEntityRecordEdits`?

I think it could, yes. I might want to totally bypass local edits, and just save something outright.

That being said, I could see an argument to be made to your latter point that it might make sense as a default, or to merge pending edits into what is passed to `saveEntityRecord`, though it's unclear whether those edits would be preferred over or defaulted in absence of what's passed explicitly to `saveEntityRecord`.

Reply:
I think this goes in conjunction with the result of `getEntityRecord`. If by default it returns the entity wiith the edits, save should also take the edits into account otherwise it's going to be confusing for third-party usage: trigger save and get a different version from the one visible before the save.

My opinion is still to keep this explicit (not bake edits) because that's how it works now otherwise we might risk some BC concerns in plugins.

Reply:
>defaulted in absence of what's passed explicitly to saveEntityRecord.

I think that would avoid BC issues right?

Reply:
It would yes, but don't you think it's weird that `getEntiityRecord` would return the persisted value by default, calling `saveEntityRecord` would bring new changes and the user might be confused (where are they coming from)?

Reply:
Maybe it's better if we have a separate action called `saveEntityRecordEdits`?